apps/settings/js/icc_menu.js

```diff
@@ -104,6 +119,18 @@
 
         iccMainHeader.hidden = false;
         iccEntries.hidden = false;
+
+        // STK Menu is ready, so we need to check if the user is in the old
+        // menu panel in order to update its content
+        if (Settings.currentPanel === '#icc') {
```

Comment:
Also check if the user is inside the same "sim card" or not. Because you'll close the SIM A entries when a SIM B menu is received.